File: client-scalaz/src/main/scala/org/openapitools/client/api/CategoryResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import CategoryResponse._

case class CategoryResponse (
  categoryId: Option[Long],
name: Option[String],
display: Option[String],
subCategoryCount: Option[Long],
parentId: Option[Long],
parentName: Option[String],
description: Option[String],
active: Option[Boolean],
asset: Option[AssetShortResponse],
favoriteId: Option[Long],
favorite: Option[Boolean],
applicationId: Option[Long],
`type`: Option[String],
externalId: Option[String],
externalType: Option[String])

object CategoryResponse {
  import DateTimeCodecs._

  implicit val CategoryResponseCodecJson: CodecJson[CategoryResponse] = CodecJson.derive[CategoryResponse]
  implicit val CategoryResponseDecoder: EntityDecoder[CategoryResponse] = jsonOf[CategoryResponse]
  implicit val CategoryResponseEncoder: EntityEncoder[CategoryResponse] = jsonEncoderOf[CategoryResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/category_response.clj
(ns sirqul-io-t-platform.specs.category-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.asset-short-response :refer :all]
            )
  (:import (java.io File)))


(def category-response-data
  {
   (ds/opt :categoryId) int?
   (ds/opt :name) string?
   (ds/opt :display) string?
   (ds/opt :subCategoryCount) int?
   (ds/opt :parentId) int?
   (ds/opt :parentName) string?
   (ds/opt :description) string?
   (ds/opt :active) boolean?
   (ds/opt :asset) asset-short-response-spec
   (ds/opt :favoriteId) int?
   (ds/opt :favorite) boolean?
   (ds/opt :applicationId) int?
   (ds/opt :type) string?
   (ds/opt :externalId) string?
   (ds/opt :externalType) string?
   })

(def category-response-spec
  (ds/spec
    {:name ::category-response
     :spec category-response-data}))
